Src/Newtonsoft.Json/JsonReader.Async.cs

```diff
@@ -105,6 +107,27 @@ public virtual Task<byte[]> ReadAsBytesAsync(CancellationToken cancellationToken
             return cancellationToken.CancelIfRequestedAsync<byte[]>() ?? Task.FromResult(ReadAsBytes());
         }
 
+
+        internal async Task<byte[]> ReadArrayIntoByteArrayAsync(CancellationToken cancellationToken)
+        {
+            List<byte> buffer = new List<byte>();
+
+            for (;;)
```

Comment:
Good enough for Kernighan & Ritchie, good enough for me. (That's a lie, I used to use their indent style once, but I dislike it now). In any case, those have all been changed to `while (true)`.